dialects/hir/src/transforms/spill/tests.rs

```diff
@@ -0,0 +1,407 @@
+use alloc::{boxed::Box, rc::Rc, sync::Arc};
+use std::string::ToString;
+
+use midenc_dialect_arith::ArithOpBuilder;
+use midenc_dialect_cf::ControlFlowOpBuilder as Cf;
+use midenc_expect_test::expect;
+use midenc_hir::{
+    dialects::builtin::{BuiltinOpBuilder, Function, FunctionBuilder},
+    pass::{Nesting, PassManager},
+    AbiParam, AddressSpace, Builder, Context, Ident, Op, OpBuilder, PointerType, ProgramPoint,
+    Report, Signature, SourceSpan, SymbolTable, Type, ValueRef,
+};
+
+use crate::{transforms::TransformSpills, HirOpBuilder};
+
+type TestResult<T> = Result<T, Report>;
+
+/// Build a simple single-block function which triggers spills and reloads,
+/// then run the `TransformSpills` pass and check that spills/reloads are
+/// materialized as `hir.store_local`/`hir.load_local`.
+#[test]
+fn materializes_spills_intra_block() -> TestResult<()> {
+    let _ = env_logger::Builder::from_env("MIDENC_TRACE")
+        .format_timestamp(None)
+        .is_test(true)
+        .try_init();
+
+    let span = SourceSpan::UNKNOWN;
+    let context = Rc::new(Context::default());
+    let mut ob = OpBuilder::new(context.clone());
+
+    let mut module = ob.create_module(Ident::with_empty_span("test".into()))?;
+    let module_body = module.borrow().body().as_region_ref();
+    ob.create_block(module_body, None, &[]);
+    let func = ob.create_function(
+        Ident::with_empty_span("test::spill".into()),
+        Signature::new(
+            [AbiParam::new(Type::Ptr(Arc::new(PointerType::new_with_address_space(
+                Type::U8,
+                AddressSpace::Element,
+            ))))],
+            [AbiParam::new(Type::U32)],
+        ),
+    )?;
+    module.borrow_mut().symbol_manager_mut().insert_new(func, ProgramPoint::Invalid);
```

Comment:
This isn't required anymore with the symbol management changes made by @lima-limon-inc. Adding/removing an `Operation` that implements `Symbol` from a `Block` in a region of an `Operation` that implements `SymbolTable`, automatically adds/removes the symbol to/from the table.

If that doesn't work, we have a bug and need to figure out why that isn't happening.

Reply:
Nice! Done in https://github.com/0xMiden/compiler/pull/636/commits/9acba821b13efe2a417652092992f66dcc8c7130